plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -504,10 +522,167 @@ export class PersonState {
+                person,
+            ]
         })
     }
 
+    private async addPersonOverride(
+        oldPerson: Person,
+        overridePerson: Person,
+        client?: PoolClient
+    ): Promise<ProducerRecord> {
+        const mergedAt = DateTime.now()
+        const oldestEvent = overridePerson.created_at
+        /**
+            We'll need to do 4 updates:
+
+         1. Add the persons involved to the helper table (2 of them)
+         2. Add an override from oldPerson to override person
+         3. Update any entries that have oldPerson as the override person to now also point to the new override person. Note that we don't update `oldest_event`, because it's a heuristic (used to optimise squashing) tied to the old_person and nothing changed about the old_person who's events need to get squashed.
+
+            TODO: how do we want to be updating oldest_event?
+
+         I'm thinking: we write it if it's not there, but don't update it on conflicts (alternative update to the older date)
+         In the transitive one the same don't update or update to older data (if that's not too complex to do)
+         it's an optimization anyway and squash job might be updating those values anyway and it's just a hint, so ignoring if complex seems better
+         in any case it's important that we use the oldest_event we stored in postgres in CH too.
+         */
+        const oldPersonId = await this.addPersonOverrideMapping(oldPerson, client)
+        const overridePersonId = await this.addPersonOverrideMapping(overridePerson, client)
+
+        await this.db.postgresQuery(
+            SQL`
+                INSERT INTO posthog_personoverride (
+                    team_id,
+                    old_person_id,
+                    override_person_id,
+                    oldest_event,
+                    version
+                ) VALUES (
+                    ${this.teamId},
+                    ${oldPersonId},
+                    ${overridePersonId},
+                    ${oldestEvent},
+                    0
+                )
+            `,
+            undefined,
+            'personOverride',
+            client
+        )
+
+        const { rows: transitiveUpdates } = await this.db.postgresQuery(
+            SQL`
+                WITH updated_ids AS (
+                    UPDATE
+                        posthog_personoverride
+                    SET
+                        override_person_id = ${overridePersonId}, version = COALESCE(version, 0)::numeric + 1
+                    WHERE
+                        team_id = ${this.teamId} AND override_person_id = ${oldPersonId}
+                    RETURNING
+                        old_person_id,
+                        version,
+                    oldest_event
+                )
+                -- The follow-up JOIN is required as ClickHouse requires UUIDs, so we need to fetch the UUIDs
```

Comment:
maybe this one also before the query

Reply:
Same comment as before, I'm neutral about this.